linkerd/meshtls/boring/src/creds/verify.rs

```diff
@@ -0,0 +1,21 @@
+use boring::x509::X509;
+use linkerd_dns_name as dns;
+use linkerd_identity as id;
+use std::io;
+
+pub(crate) fn verify_id(cert: &X509, id: &id::Id) -> io::Result<()> {
+    for san in cert.subject_alt_names().into_iter().flatten() {
+        if let Some(n) = san.dnsname() {
+            if let Ok(name) = n.parse::<dns::Name>() {
```

Comment:
if the value we get from `san.dnsname()` doesn't parse as a DNS name, it looks like we just skip it...should we maybe log a warning when we skip a subject alternate name "DNS name" that didn't parse as a valid DNS name?